src/DynamoCoreWpf/Views/Core/WorkspaceView.xaml.cs

```diff
@@ -313,99 +314,144 @@ internal Rect GetVisibleBounds()
-            bounds.Width = 20 + ((((int)Math.Ceiling(bounds.Width)) + 1) & ~0x01);
-            bounds.Height = 20 + ((((int)Math.Ceiling(bounds.Height)) + 1) & ~0x01);
+                var currentTransformGroup = WorkspaceElements.RenderTransform as TransformGroup;
+                WorkspaceElements.RenderTransform = new TranslateTransform(10.0 - bounds.X - minX, 10.0 - bounds.Y - minY);
+                WorkspaceElements.UpdateLayout();
 
-            var currentTransformGroup = WorkspaceElements.RenderTransform as TransformGroup;
-            WorkspaceElements.RenderTransform = new TranslateTransform(10.0 - bounds.X - minX, 10.0 - bounds.Y - minY);
-            WorkspaceElements.UpdateLayout();
+                rtb = new RenderTargetBitmap(((int)bounds.Width),
+                    ((int)bounds.Height), 96, 96, PixelFormats.Default);
 
-            var rtb = new RenderTargetBitmap(((int)bounds.Width),
-                ((int)bounds.Height), 96, 96, PixelFormats.Default);
+                rtb.Render(WorkspaceElements);
+                WorkspaceElements.RenderTransform = currentTransformGroup;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+            return rtb;
+        }
+
+        /// <summary>
+        /// Function to validate if the Workspace is valid for export as image / save it as image
+        /// </summary>
+        /// <param name="validating"></param>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        [HandleProcessCorruptedStateExceptions]
+        internal ExportImageResult IsWorkSpaceRenderValidAsImage(bool validating, string path = null)
+        {
+            ExportImageResult result = ExportImageResult.EmptyDrawing;
+            RenderTargetBitmap workSpaceRender = GetRender();
+            if (workSpaceRender == null) return result;
 
-            rtb.Render(WorkspaceElements);
-            WorkspaceElements.RenderTransform = currentTransformGroup;
+            result = ExportImageResult.IsValidAsImage;
+            if (validating) path = this.ViewModel.FileName.Replace(".dyn", ".png");
 
             try
             {
                 using (var stm = System.IO.File.Create(path))
                 {
                     // Encode as PNG format
                     var pngEncoder = new PngBitmapEncoder();
-                    pngEncoder.Frames.Add(BitmapFrame.Create(rtb));
+                    pngEncoder.Frames.Add(BitmapFrame.Create(workSpaceRender));
                     pngEncoder.Save(stm);
                 }
             }
+            catch (AccessViolationException)
+            {
+                result = ExportImageResult.NotValidAsImage;
+            }
             catch (Exception)
             {
+                result = ExportImageResult.NotValidAsImage;
+            }
+
+            if (validating && System.IO.File.Exists(path))
+            {
+                System.IO.File.Delete(path);
```

Comment:
Can you put comments for the deletion code? Also, would delete always succeed, any case user would not have permission?

Reply:
> Can you put comments for the deletion code? Also, would delete always succeed, any case user would not have permission?

Done. I'm taking advantage of the BackupLocation since it's already in a safe place https://github.com/DynamoDS/Dynamo/pull/14055/commits/9ef500c34f73bf8c829d95e6323d84be6941fbfa